### ISSUE TITLE
Add compression_format = inherit

### ISSUE THREAD

**Issue by cgwalters** [containers/buildah#5586]
Related to https://github.com/containers/common/pull/2048 

How about adding `compression_format = inherit`?  The semantics would be that we default compression for new layers to previous base image layer by default.  IOW, we default to e.g. `zstd:chunked` only if we detect at least one usage of it in a parent layer.

That would *greatly* reduce the "flag day" effect of this and allow individual image maintainers to opt in as they desire.

**Comment by cgwalters** [comment]
Referenced this again in context of https://fedoraproject.org/wiki/Changes/zstd:chunked where I think this would still help a lot versus us trying to roll out a global flag to turn it on for all image pushing.

**Comment by nalind** [comment]
With "inherit", the compression used for an image that did not reuse layers from a base image would still need to be decided somehow.

At push-time, given that we write manifests for new images using uncompressed MediaType values (unless buildah's been told to compress it, something which has historically been strongly discouraged), the manifest can't be used as a source for that information.

At commit-time, buildah could examine the `CompressionType` that's recorded with the base image's layers to try to figure out how they were originally compressed, but since it might also have built that image, it would have to keep walking up the layers list until it found a value other than "uncompressed" to find a good answer.  But then, this logic could be (and would probably need to be) deferred to push-time.

**Comment by nalind** [comment]
Wait, is the logic for `compression_format` only implemented in podman?

**Comment by cgwalters** [comment]
> With "inherit", the compression used for an image that did not reuse layers from a base image would still need to be decided somehow.

Yes, no argument there. But the set of folks building from-scratch images is relatively small and I think can be assumed to have some sophistication.

> At push-time, given that we write manifests for new images using uncompressed MediaType values (unless buildah's been told to compress it, something which has historically been strongly discouraged), the manifest can't be used as a source for that information.

Right, but we could store this data in some lookaside location during the build.

There's also a somewhat related topic in that if we did that, we could also make this a *build time* property and not a *push time* property, which might align honestly with how a lot of people tend to think of these things. Yes, the actual generation of compressed layers may continue to happen on push...but this is an important property of an image, and one would tend to expend to find "important image properties" around the logic for where it's "built", not where it's pushed.

**Comment by mtrmac** [comment]
Thinking about the UI … there’s also the “publish variants with two compression algorithms” case, where there is a single c/storage image, a single on-registry destination _name_, but multiple destination formats. I suppose if we had a _build-time_ “and compress this way when publishing” property, this should be also included (?!).
